ugc/ugc_tests/storage_tests.cpp

```diff
@@ -0,0 +1,253 @@
+
+  static MwmBuilder & Builder()
+  {
+    static MwmBuilder builder;
+    return builder;
+  }
+
+  void Build(vector<TestPOI> const & poi)
+  {
+    BuildMwm([&poi](TestMwmBuilder & builder)
+             {
+               for (auto const & p : poi)
+                 builder.Add(p);
+             });
+  }
+
+  FeatureID FeatureIdForCafeAtPoint(m2::PointD const & mercator)
+  {
+    return FeatureIdForPoint(mercator, ftypes::IsFoodChecker::Instance());
+  }
+
+  FeatureID FeatureIdForRailwayAtPoint(m2::PointD const & mercator)
+  {
+    return FeatureIdForPoint(mercator, ftypes::IsRailwayStationChecker::Instance());
+  }
+
+  Index & GetIndex() { return m_index; }
+
+private:
+  MwmBuilder()
+  {
+    classificator::Load();
+  }
+
+  MwmSet::MwmId BuildMwm(BuilderFn const & fn)
+  {
+    if (m_testMwm.OnDisk(MapOptions::Map))
+      Cleanup(m_testMwm);
+
+    m_testMwm = platform::LocalCountryFile(GetPlatform().WritableDir(),
+                                           platform::CountryFile(kTestMwmName),
+                                           0 /* version */);
+    {
+      generator::tests_support::TestMwmBuilder builder(m_testMwm, feature::DataHeader::country);
+      fn(builder);
+    }
+
+    auto result = m_index.RegisterMap(m_testMwm);
+    CHECK_EQUAL(result.second, MwmSet::RegResult::Success, ());
+
+    auto const & id = result.first;
+
+    auto const & info = id.GetInfo();
+    if (info)
+      m_infoGetter.AddCountry(storage::CountryDef(kTestMwmName, info->m_limitRect));
+
+    CHECK(id.IsAlive(), ());
+    return id;
+  }
+
+  template<typename Checker>
+  FeatureID FeatureIdForPoint(m2::PointD const & mercator, Checker const & checker)
+  {
+    m2::RectD const rect = MercatorBounds::RectByCenterXYAndSizeInMeters(mercator, 0.2 /* rect width */);
+    FeatureID id;
+    auto const fn = [&id, &checker](FeatureType const & featureType)
+    {
+      if (checker(featureType))
+        id = featureType.GetID();
+    };
+    m_index.ForEachInRect(fn, rect, scales::GetUpperScale());
+    CHECK(id.IsValid(), ());
+    return id;
+  }
+
+  void Cleanup(platform::LocalCountryFile const & map)
+  {
+    m_index.DeregisterMap(map.GetCountryFile());
+    platform::CountryIndexes::DeleteFromDisk(map);
+    map.DeleteFromDisk(MapOptions::Map);
+  }
+
+  Index m_index;
+  storage::CountryInfoGetterForTesting m_infoGetter;
+  platform::LocalCountryFile m_testMwm;
+};
+}  // namespace ugc_tests
+
+using namespace ugc_tests;
+
+UNIT_TEST(StorageTests_Smoke)
+{
+  auto & builder = MwmBuilder::Builder();
+  m2::PointD const point(1.0, 1.0);
+  builder.Build({TestCafe(point)});
+  auto const id = builder.FeatureIdForCafeAtPoint(point);
+  auto const original = MakeTestUGCUpdate(Time(chrono::hours(24 * 300)));
+  Storage storage(builder.GetIndex());
+  storage.SetUGCUpdate(id, original);
+  auto const actual = storage.GetUGCUpdate(id);
+  TEST_EQUAL(original, actual, ());
+  TEST(!storage.GetUGCToSend().empty(), ());
```

Comment:
Is it possible to check the content too?

Reply:
Thanks.
Will add this unit test within test period.